pkg/discovery/discovery_client.go

```diff
@@ -162,15 +163,31 @@ func (discoveryClient *MesosDiscoveryClient) Validate(accountValues []*proto.Acc
 // Discover the Target Topology
 func (discoveryClient *MesosDiscoveryClient) Discover(accountValues []*proto.AccountValue) (*proto.DiscoveryResponse, error) {
 	// TODO: use account values to create target conf ?
+	//target, err := conf.CreateMesosTargetConf(discoveryClient.mesosMasterType, accountValues)
+	//mesosConf, ok := target.(*conf.MesosTargetConf)
+	//if (!ok) {
+	//	return nil, fmt.Errorf("Invalid target : %s", accountValues)
+	//}
+
+	mesosConf := discoveryClient.clientConf
+	// Select the leader to connect
+	if discoveryClient.urlList == nil {
+		discoveryClient.urlList = ParseMasterIPPorts(mesosConf.MasterIPPort)
+	}
+	//urlList := ParseMasterIPPorts(mesosConf.MasterIPPort)
+	mesosLeader, err := NewMesosLeader(mesosConf.Master, discoveryClient.urlList , mesosConf.MasterUsername, mesosConf.MasterPassword)
```

Comment:
this means that the items of discoveryClient.urlList should be accessed in a order that current.leader should be first.